src/v2/components/PotentialFireList.jsx

```diff
@@ -90,19 +91,28 @@ export default function PotentialFireList() {
     }
 
     const {current: allFires} = allFiresRef
+    const {current: firesByKey} = firesByKeyRef
     const key = getCameraKey(fire)
-    const existingFire = allFires.find((x) => hasCameraKey(x, key))
 
-    if (existingFire != null) {
-      if (existingFire.croppedUrl !== croppedUrl) {
+    if (firesByKey[key] != null) {
+      if (firesByKey[key].croppedUrl !== croppedUrl) {
         // Fire is already indexed, but its video has been updated.
-        existingFire.croppedUrl = croppedUrl
+        firesByKey[key].croppedUrl = croppedUrl
         // TODO: Update video source, reload video.
         console.error('Not implemented: updateFires()')
       }
     } else {
+      firesByKey[key] = fire
+
       allFires.unshift(fire)
       allFires.sort((a, b) => b.sortId - a.sortId)
+
+      allFires.forEach((a) => {
+        allFires.forEach((b) => {
+          hasAngleOfFire(a, b)
```

Comment:
The only way I see of odd stuff happening here requiring the double loop is if sortID and timestamp get flipped.  I wonder if hasAngleOfFire should not use timestamps at all...

Reply:
It’d be nice if events streamed in order. I don’t know that this has ever been the case. i.e., we can see incoming fire events being sorted in the old/existing code (by `sortId` descending), which is why I included the sort in v2:

https://github.com/open-climate-tech/checkfire/blob/master/src/VoteFires.js#L241

Somewhat unexpectedly, the server-side events are already sorted by `sortId` descending:

https://github.com/open-climate-tech/checkfire/blob/master/server-src/sse.js#L87

Then reversed?

https://github.com/open-climate-tech/checkfire/blob/master/server-src/sse.js#L89

As for not using `timestamp` at all… How would we know which events fall within the allotted time period? That’s the only thing `hasAngleOfFire()` is using `timestamp` for…